src/fetcher.ts

```diff
@@ -215,4 +245,62 @@ export abstract class Fetcher {
     const feeReceiver = await factoryContract.feeTo()
     return { feeDenominator, feeReceiver }
   }
+  /**
+   * Fetches user created factory pools for curve protocol
+   */
+
+  public static async fetchCurveFactoryPools(chainId: ChainId): Promise<CurvePool[]> {
+    if (CURVE_FACTORY_SUPPORTED_APIS[chainId] === '') return []
+
+    const response = await fetch(`https://api.curve.fi/api/getPools/${CURVE_FACTORY_SUPPORTED_APIS[chainId]}/factory`)
+
+    if (!response.ok) throw new Error('response not ok')
+    const allPoolsArray = (await response.json()) as FactoryPoolsApiResponse
+    //filter for low liquidty pool
+    const filteredLowLiquidityPools = allPoolsArray.data.poolData.filter((item) => item.usdTotal > 100000)
+    //restructures pools so they fit into curvePool type
+    const modifiedArray: CurvePool[] = filteredLowLiquidityPools.map(
+      ({ symbol, name, coins, address, implementation, isMetaPool }) => {
+        const tokens: CurveToken[] = coins.map((token) => {
+          let currentToken = new Token(chainId, token.address, parseInt(token.decimals), token.symbol, token.name)
+
+          //wraps token if its Native so that it can be matched
+          if (token.address === '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE')
+            currentToken = Token.getNativeWrapper(chainId)
```

Comment:
Some pools use the native token _address_ intentionally to indicate that the pool allows trading the native assets. Does this cover such case?

Reply:
It doesn't but if you actually put the native token it and its not going through smart router current implementation won't pick it up because of how we hande native pair sorting...I made an issue about it and can tackle it in another pr.
https://github.com/levelkdev/swapr-sdk/issues/174 @adamazad wdyt?

Reply:
Yes, another PR makes sense